src/hooks/useUserAccessConfig.js

```diff
@@ -0,0 +1,212 @@
+            return {
+              ...prevErrors,
+              [name]: ""
+            };
+          });
+        })
+        .catch((error) => {
+          setErrors((prevErrors) => {
+            return {
+              ...prevErrors,
+              [name]: error.message
+            };
+          });
+        });
+    },
+    [schema]
+  );
+  const validateObject = useCallback(
+    (formData) => {
+      schema
+        .validate(formData)
+        .then(() => {
+          setErrors({});
+        })
+        .catch((error) => {
+          setErrors((prevErrors) => {
+            return {
+              ...prevErrors,
+              ...error
+            };
+          });
+        });
+    },
+    [schema]
+  );
+
+  const handleChange = useCallback(
+    (event) => {
+      const { name, value } = event.target;
+      setFormData((prevPageConfig) => {
+        return {
+          ...prevPageConfig,
+          [name]: value
+        };
+      });
+      validateField(name, value);
+    },
+    [validateField, setFormData]
+  );
+  const saveAccessConfig = useCallback(() => {
+    validateObject(formData);
+    axiosPrivate
+      .post(`/api/protected/${currentConfig.userProfileAccessConfig}`, formData)
+      .then(({ data }) => {
+        console.log("saved successfully ", data);
+      })
+      .catch((err) => {
+        console.error("error :>> ", err);
+      });
+  }, [formData, validateObject]);
+  const handleTabChange = useCallback(
+    (event, newValue) => {
+      console.log("🚀 ~ useUserAccessConfig ~ newValue:", newValue);
+      setActiveTab(newValue);
+    },
+    [setActiveTab]
+  );
+
+  const handleAddModalOpen = useCallback((title) => {
+    setPageDialogTitle(title);
+    setModalOpenAdd(() => true);
+  }, []);
+  const handleAddModalClose = useCallback(() => {
+    setModalOpenAdd(false);
+    setFormData(pageConfigInitial);
+  }, [setModalOpenAdd, setFormData]);
+
+  const updateAccessConfig = useCallback(() => {
+    validateObject(formData);
+    const { access, accessListChanged = {}, ...formattedData } = formData;
+    const accessListPrevious = {};
+    // Formatting access array
+    access.map((prevAccessConfig) => {
+      Object.entries(prevAccessConfig).forEach(([key, value]) => {
+        accessListPrevious[key] = value;
+      });
+    });
+    // Formatting access array
+    formattedData.access = Object.entries({ ...accessListPrevious, ...accessListChanged }).map(([key, value]) => {
+      return { [key]: value };
+    });
+
+    axiosPrivate
+      .put(`/api/protected/${currentConfig.userProfileAccessConfig}`, formattedData)
+      .then(({ data }) => {
+        console.log("saved successfully ", data);
+      })
+      .catch((err) => {
+        console.error("error :>> ", err);
+      });
+    // setFormData(pageConfigInitial);
+  }, [validateObject, formData]);
+
+  const populateAccessConfigForm = useCallback(
+    (row) => {
+      console.log("{row} :>> ", { row });
+      axiosPrivate
```

Comment:
you are using different approach , you are calling get api  ,then set the formData ,  why you are not using  `row.original` to set the form data .

Reply:
row data is insufficient so need to use api response data